CenterForOpenScience/osf.io
# [ENG-2635] DraftNodeDraftRegistration Endpoint

## Purpose
These changes are to make it easier for our front-end to use the API for No-Project Registrations.

These changes are based off of the changes in [this PR](https://github.com/CenterForOpenScience/osf.io/pull/9621)

## Changes
* Add a DraftRegistration relationship to the DraftNode serializer
	* Create a DraftNodeDraftRregistrationsList Endpoint
	* Create a relationship to the DraftNodeDraftRregistrationsList Endpoint
	* Adds tests for the DraftNodeDraftRegistrationsList
* Fix `type` metadata of the `branched_from` relationship field on the DraftRegistration serializer to reflect when a draft registration is branched from a DraftNode

## QA Notes
Please make verification statements inspired by your code and what your code touches.
- Verify that the new DraftNodeDraftRegistration list endpoint returns the sole draft registration associated with the draft node 
- Verify that the `type` field of the `branched_from` relationship for DraftRegistrations is accurate for draft registrations based off of projects (nodes) and draft nodes. 

What are the areas of risk? N/A

Any concerns/considerations/questions that development raised?
- We opted to go with a 'list' of draft registrations associated with a draft node for the API endpoint although there should only ever be a single draft registration associated with a draft node to make front-end ingestion of the API easier/use existing logic.

## Documentation
Our API documentation for DraftNodes should be updated to reflect the addition of the new endpoint

## Side Effects
N/A

## Ticket
[JIRA Ticket](https://openscience.atlassian.net/browse/ENG-2635)
